comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -366,20 +366,47 @@ func (r *secretResolver) Resolve(data []byte, origin string) ([]byte, error) {
 	return finalConfig, nil
 }
 
-// allowlistPaths restricts what config settings may be updated
-// tests can override this to exercise functionality: by setting this to nil, allow all settings
-// NOTE: Related feature to `authorizedConfigPathsCore` in `comp/api/api/apiimpl/internal/config/endpoint.go`
-var allowlistPaths = map[string]struct{}{"api_key": {}, "app_key": {}, "external_metrics_provider/api_key": {}, "external_metrics_provider/app_key": {}}
+// allowlistPaths restricts what config settings may be updated. Any secrets linked to a settings containing any of the
+// following strings will be refreshed.
+//
+// For example, allowing "additional_endpoints" will trigger notifications for:
+//   - "additional_endpoints"
+//   - "logs_config.additional_endpoints"
+//   - "logs_config.additional_endpoints.url"
+//   - ...
+//
+// NOTE: Related feature to `authorizedConfigPathsCore` in `comp/api/api/def/component.go`
+var (
+	allowlistPaths = []string{
+		"api_key",
+		"app_key",
+		"additional_endpoints",
+	}
+	// tests override this to test refresh logic
+	allowlistEnabled = true
+)
+
+func secretMatchesAllowlist(secretCtx secretContext) bool {
+	if !allowlistEnabled {
+		return true
+	}
```

Comment:
We are checking here an in line 405. Do we need both checks?

Reply:
Yes, this function is used directly line 441.

Reply:
I find weird that we have two conditions to see if we can use the allow list: 

```
useAllowlist && allowlistEnabled
```

Could it be simplified?

Reply:
Thanks for the explanation